Add apigate proxy response

minsikzzang opened the pull request:
* Add Definitions for Api Gateway Custom Authorizer request and response

More detail
* http://docs.aws.amazon.com/apigateway/latest/developerguide/use-custom-authorizer.html

minsikzzang committed:
Add apigate proxy response

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	// PUT.
+	HTTPMethod string
+
+	// The incoming reauest HTTP headers.
```
Typo :wink:

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+
+// RequestContext provides contextual information about an Amazon API Gateway
+// Proxy Custom Authorizer event.
+//
```
plz delete this empty comment line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	Type string
+
+	// MethodARN is the ARN of the incoming method request and is populated by
+	// API Gateway in accordance with the custom authorizer configuration
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+// Event represents an input to a REQUEST authorizer for an API method with a
+// proxy integration
+type Event struct {
+	// Type property specifies the authorizer type ("TOKEN" or "REQUEST")
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	Resource string
+}
+
+// PolicyDocument represents AWS IAM policy document
```
... represents **an** AWS IAM ....

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+type PolicyDocument struct {
+	Version string
+
+	// Statement represents AWS IAM policy statement
```
... represents **an** AWS IAM ....

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+}
+
+// Event represents an input to a REQUEST authorizer for an API method with a
+// proxy integration
```
plz  add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	PolicyDocument PolicyDocument `json:"policyDocument"`
+
+	// API Gateway passes the context object from a custom authorizer directly to
+	// the backend Lambda function as part of the input event
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	// The principal user identification associated with the token sent by the client.
+	PrincipalID string `json:"principalId"`
+
+	// PolicyDocument represents AWS IAM policy document
```
plz add "." at end of line and change to "...represents **an** AWS..."

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+	Statement []Statement
+}
+
+// Response represents an output from an Amazon API Gateway Custom Authorizer
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/doc.go:
```diff
@@ -0,0 +1,21 @@
+//
+
+/*
+Package apigatewayproxyevt allows you write AWS Lambda functions as the back end
```
Change for correct description. Like:
Package apigatewayproxyevt allows you to write an Amazon API Gateway custom authorizer that provide control access to your API methods.

lion3ls commented on service/lambda/runtime/event/apigatewayproxyevt/definition.go:
```diff
@@ -151,6 +151,24 @@ type Event struct {
 	RequestContext *RequestContext
 }
 
+// Response represents a API Gateway Lambda Proxy response format
```
plz add "." at end of line, and "...represents **an** API..."

lion3ls commented on service/lambda/runtime/event/apigatewayproxyevt/definition.go:
```diff
@@ -151,6 +151,24 @@ type Event struct {
+	// encoded.
+	IsBase64Encoded bool `json:"isBase64Encoded"`
+
+	// The outgoing HTTP status code
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+
+	// AuthorizationToken has called supplied token which is allow, deny,
+	// unauthorized, or any other string value. An empty string value is the
+	// same as unauthorized (only for TOKEN type)
```
plz add "." at end of line

lion3ls commented on service/lambda/runtime/event/apigatewayproxyevt/definition.go:
```diff
@@ -151,6 +151,24 @@ type Event struct {
 }
 
+// Response represents a API Gateway Lambda Proxy response format
+type Response struct {
```
To preserve the good consistency of PR, please make this change in an other PR.

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+// RequestContext provides contextual information about an Amazon API Gateway
+// Proxy Custom Authorizer event.
+//
+type RequestContext struct {
```
The parameter "RequestID" is missing.
```json
    "requestContext": {
        "path": "/request",
        "accountId": "123456789012",
        "resourceId": "05c7jb",
        "stage": "test",
        "requestId": "...",
        "identity": {
            "apiKey": "...",
            "sourceIp": "..."
        },
        "resourcePath": "/request",
        "httpMethod": "GET",
        "apiId": "s4x3opwd6i"
    }
```
Please add it with a short description.

cf. [doc](http://docs.aws.amazon.com/apigateway/latest/developerguide/use-custom-authorizer.html)

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+
+// PolicyDocument represents AWS IAM policy document
+type PolicyDocument struct {
+	Version string
```
Please add description.

lion3ls commented on service/lambda/runtime/event/apigatewayauthorizerevt/definition.go:
```diff
@@ -0,0 +1,149 @@
+//
+// More detail about IAM policy statement
+// http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html#api-gateway-calling-api-permissions
+type Statement struct {
```
Please add short description for this 3.

minsikzzang committed:
Address review comments(more description / fix typoes / change styles)

minsikzzang committed:
Update description in doc.go

minsikzzang commented:
@lsuss addressed all comments, I will add another one for adding response in `apigatewayproxyevt`, and found one typo in the `apigatewayproxyevt`, so that will be corrected as well.

lsuss commented:
One last thing I just noticed now sry.

Please add README.md for apigatewayauthorizerevt  (check other events). And Update the main README.md to add link to your new supported event.

You did a great work btw :+1: you are the very first to help us by adding an event type so it's a special moment for us you know :sunglasses:.

minsikzzang committed:
Add README.md

minsikzzang committed:
Link apigatewaycustomauthroizer README to the main README.md
